File: 05-aquasentinel-fish-farm-water-quality/sensors/mqtt_publisher.py
"""Publishes a single sensor reading to its MQTT topic as JSON."""

import json


def publish_reading(mqtt_client, reading):
    """Publish `reading` dict to aquasentinel/{pondId}/{metric}.

    reading must contain pondId and metric keys per the shared contract.
    """
    topic = f"aquasentinel/{reading['pondId']}/{reading['metric']}"
    payload = json.dumps(reading)
    mqtt_client.publish(topic, payload)

File: 11-parkfog-kerbside-occupancy/sensors/mqttPublisher.js
'use strict';

// builds the exact topic pattern from the shared contract and publishes the reading as JSON
function publishReading(mqttClient, reading) {
  const topic =
    reading.scope === 'bay'
      ? `parkfog/bay/${reading.id}/${reading.metric}`
      : `parkfog/zone/${reading.id}/${reading.metric}`;

  mqttClient.publish(topic, JSON.stringify(reading));
  return topic;
}

module.exports = { publishReading };
